src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -461,14 +461,6 @@ internal async Task<(IReadOnlyList<T> results, string continuationToken)> Execut
                     if (page.Count > 0)
                     {
                         results.AddRange(page);
-                        if (mustNotExceedMaxItemCount && results.Count > feedOptions.MaxItemCount)
```

Comment:
if(executingWithMaxParallelism) probably should have been here. I still see an impact when QueryStatistics is on, but having this disabled in other cases looks right

Reply:
Also if `feedOptions.MaxItemCount` was being changed, it should have used the captured `totalDesiredCount`